types/src/transaction/mod.rs

```diff
@@ -422,6 +525,49 @@ pub enum TransactionPayload {
     /// A multisig transaction that allows an owner of a multisig account to execute a pre-approved
     /// transaction as the multisig account.
     Multisig(Multisig),
+
+    /// Question: Any better name for this variant?
+    V2(TransactionPayloadV2),
```

Comment:
thinking about how to use this nicely, maybe we should deprecate payload() function (i.e. remove it / rename to payload_deprecated), and instead provide these functions:

executable() - which for V2 returns executable, for others, extracts Script/EntryFunction, and createsExecutable from it
multisig_address() - which gives it from either V2 or Multisig variant
replay_protector() - which you have

and then from outside , it looks like TransactionPayload doesn't even exist, nobody matches on V2, we can even rename it to TransactionPayloadWrapper or something and have this V2 be named Direct. We don't expect ever to need v2, changes will go to TransactionExecutable/TransactionExtraConfig. 

only issue might be how to do that without too much cloning for executable() call - maybe adding some non-serializable field that keeps the wrapper when accessed first, like we do for hash - maybe others have better ideas